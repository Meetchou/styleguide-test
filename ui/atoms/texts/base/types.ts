export type HProps = {
  className?: string,
  children: React.ReactNode;
};
