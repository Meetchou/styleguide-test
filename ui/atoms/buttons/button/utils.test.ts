import { ColorType, SizeType } from '@ui/types';

import {
  getButtonCls,
  sizes,
  buttonTheme
} from './utils';

describe('atoms/buttons/button/utils', () => {
  describe('getButtonCls', () => {
    it('return by default', () => {
      expect(getButtonCls()).toMatchSnapshot();
    });

    it('should add className', () => {
      const className = 'this-is-a-test';
      expect(getButtonCls({ className })).toContain(className);
    });

    describe('should override cursor when args', () => {
      it.each([
        { saving: true },
        { disabled: true },
      ])('%s', args => {
        expect(getButtonCls(args)).toContain('cursor-default');
      });
    });

    describe('size params implement right size', () => {
      it.each(Object.keys(sizes) as SizeType[])('%s', size => {
        expect(getButtonCls({ size })).toContain(sizes[size]);
      });
    });

    describe('themes should be applied', () => {
      it.each(Object.keys(buttonTheme) as ColorType[])('%s', color => {
        expect(getButtonCls({ color })).toContain(buttonTheme[color]);
      });
    });
  });
});
