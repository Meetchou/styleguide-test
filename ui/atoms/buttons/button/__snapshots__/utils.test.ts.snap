// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/buttons/button/utils getButtonCls return by default 1`] = `"relative text-center font-semibold transition-all text-sm rounded-lg border outline-none transition-colors flex items-center justify-center text-primary-500 bg-white border-gray-100 hover:bg-primary-100 hover:border-primary-100 h-9"`;
